tests/test_util/test_solver.py

```diff
@@ -118,14 +118,19 @@ def test_absolute_expression(model: "Model") -> None:
 def test_fix_objective_as_constraint(solver: str, model: "Model") -> None:
     """Test fixing present objective as a constraint."""
     model.solver = solver
+    opt = model.slim_optimize()
     with model as m:
         su.fix_objective_as_constraint(model, 1.0)
         constraint_name = m.constraints[-1]
-        assert abs(m.constraints[-1].expression - m.objective.expression) < 1e-6
+        assert (m.constraints[-1].expression - m.objective.expression).simplify() == 0
+        assert m.constraints[-1].lb == pytest.approx(opt)
```

Comment:
It seems better to me, to pass in a name for the constraint, and then use that name to retrieve the constraint object.

Reply:
Also, the variable name `constraint_name` is misleading here, as it is in fact the object and not the name.

Reply:
Yeah, I did not write the original test and did not want to modify more than necessary. I changed it now to be as suggested.